tests/app/notify_client/test_service_api_client.py

```diff
@@ -541,14 +541,14 @@ def test_client_deletes_service_template_cache_when_service_is_updated(
     mocker.patch('app.notify_client.current_user', id='1')
     mocker.patch('notifications_python_client.base.BaseAPIClient.request')
     mock_redis_delete = mocker.patch('app.extensions.RedisClient.delete')
+    mock_redis_delete_by_pattern = mocker.patch('app.extensions.RedisClient.delete_cache_keys_by_pattern')
 
     service_api_client.update_reply_to_email_address(SERVICE_ONE_ID, uuid4(), 'foo@bar.com')
 
-    assert len(mock_redis_delete.call_args_list) == 2
-    assert mock_redis_delete.call_args_list[1] == call('service-{}'.format(SERVICE_ONE_ID))
+    assert len(mock_redis_delete.call_args_list) == 1
+    assert mock_redis_delete.call_args_list[0] == call('service-{}'.format(SERVICE_ONE_ID))
 
-    templates_to_delete = mock_redis_delete.call_args_list[0][0]
-    assert len(templates_to_delete) == 6
-    for template_key in templates_to_delete:
+    assert len(mock_redis_delete_by_pattern) == 6
```

Comment:
Test is failing because you’re trying to assert the length of the mock, not its calls
```suggestion
    assert len(mock_redis_delete_by_pattern.call_args_list) == 6
```

Reply:
doh!
Thanks